### DISCUSSION TITLE
After following the Getting started guide, exo container do not start correctly

### DISCUSSION THREAD

**Discussion by rdenarie** [exoplatform/exo-community#5] [Q&A]
I continue a discussion started by email :

After following the instructions, I always get that it exits with a code 1.  Can you help point out what would be the issue. I am using an ubuntu Server

22.04 LTS with docker and docker-compose installed.

![image](https://user-images.githubusercontent.com/807839/191216577-b266d84a-e71d-4263-8fe1-100be8c99b77.png)

**Comment by rdenarie** [comment]
I asked for exo server logs : 
`docker logs exo_home_exo_1`
results are : 
[exo installation error.txt](https://github.com/exoplatform/exo-community/files/9606040/exo.installation.error.txt)

I asked for database server logs : 
`docker logs exo_home_mysql_1`
Results : 
[error 2.txt](https://github.com/exoplatform/exo-community/files/9606057/error.2.txt)

**Comment by rdenarie** [comment]
Can you try to complety stop all containers wit this command 
`docker-compose -f docker-compose.yml down`

And then restart it with 
`docker-compose -f docker-compose.yml up`

If it doesnt work, we will try to  update the docker-compose configuration.

**Reply by BivanOdima** [reply]
This is the error I get after restarting

**exo_1    | wait-for: timeout occurred after waiting 60 seconds for mysql:3306
exo_1    | [ERROR] The mysql database mysql:3306 was not available within 60s ! eXo startup aborted ...
exo_home_exo_1 exited with code 1**

**Reply by rdenarie** [reply]
So, we will check if you are able to connect to the mysql server without eXo, with the mysql client.
If the mysqclient is not present on your machine, you can install with : 
`sudo apt-get install mysql-client`

With docker containers up, type this command : 
`docker inspect exo_home_mysql_1`
It will give you information about the container mysql. The one we search is 
> IPAddress": "192.168.96.4",

When you have found it, type this command, by replacing 192.168.96.4 with the ip you found with the previous command.
`mysql -u exo -p -h 192.168.96.4` 
It should ask for a password, it is the one in the docker compose if you havn't change it : 
>  MYSQL_PASSWORD: my-secret-pw

If you see something like this : 

>  Welcome to the MySQL monitor.  Commands end with ; or \g.
> Your MySQL connection id is 25
> Server version: 8.0.28 MySQL Community Server - GPL
> 
> Copyright (c) 2000, 2022, Oracle and/or its affiliates.
> 
> Oracle is a registered trademark of Oracle Corporation and/or its
> affiliates. Other names may be trademarks of their respective
> owners.
> 
> Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
> 
> mysql>

you are connected to the mysql server. 

Please let me know if you manage to realize this step.

**Reply by BivanOdima** [reply]
1. Result of :
`sudo apt-get install mysql-client`

**ubuntu@ubuntu:~$ sudo apt-get install mysql-client
[sudo] password for ubuntu:
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
mysql-client is already the newest version (8.0.30-0ubuntu0.22.04.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.**

2. Result of:
`docker inspect exo_home_mysql_1`

 "IPAddress": "172.19.0.3"
 
3. Reuslt of:

`mysql -u exo -p -h 172.19.0.3`


**Enter password:
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 8
Server version: 8.0.28 MySQL Community Server - GPL**

**Copyright (c) 2000, 2022, Oracle and/or its affiliates.**

**Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.**

**Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.**

**mysql>**


I have realized the step.

**Reply by rdenarie** [reply]
Hello, 
So the mysql server is available on the port 3306.
We will now check that exo and mysql are correctly on the same network.

Can you pass these commands :
`docker inspect exo_home_exo_1 | grep IPAddress`
`docker inspect exo_home_mysql_1 | grep IPAddress`

In addition, can you provide docker and docker-compose version : 
`docker --version`
`docker-compose --version`

**Reply by BivanOdima** [reply]
1. Result of:

ubuntu@ubuntu:~$ `sudo docker inspect exo_home_exo_1 | grep IPAddress`
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "",


2.Result of 


**ubuntu@ubuntu:~$ `sudo docker inspect exo_home_mysql_1 | grep IPAddress`
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "172.19.0.3",**

3. Result of 

ubuntu@ubuntu:~$ `docker --version`
Docker version 20.10.18, build b40c2f6

4.Result of

ubuntu@ubuntu:~$  `docker-compose --version`
docker-compose version 1.29.2, build 5becea4c
ubuntu@ubuntu:~$

**Reply by BivanOdima** [reply]
It looks like the exo_home_exo_1 container  doesn't have an IP Address

**Reply by rdenarie** [reply]
It is because the container is stopped. 
Can you redo the command during the startup (before the container stops) ?

**Comment by BivanOdima** [comment]
1. Result of:

ubuntu@ubuntu:~$ `sudo docker inspect exo_home_exo_1 | grep IPAddress`
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "",


2.Result of 


**ubuntu@ubuntu:~$ `sudo docker inspect exo_home_mysql_1 | grep IPAddress`
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "172.19.0.3",**

3. Result of 

ubuntu@ubuntu:~$ `docker --version`
Docker version 20.10.18, build b40c2f6

4.Result of

**ubuntu@ubuntu:~$ `docker-compose --version`
docker-compose version 1.29.2, build 5becea4c
ubuntu@ubuntu:~$**

**Reply by rdenarie** [reply]
For 1), as you exo container is stopped we dont see his ipAdress.

Can you try to down all containers
`docker-compose -f docker-compose.yml down`

And then restart it with
`docker-compose -f docker-compose.yml up`

Then during the restart, recheck the both ipAdress (before exo container stops)
```
sudo docker inspect exo_home_exo_1 | grep IPAddress
sudo docker inspect exo_home_mysql_1 | grep IPAddress
```

**Reply by BivanOdima** [reply]
1. When the exo_home container is still up. Result is

ubuntu@ubuntu:~$  `sudo docker inspect exo_home_exo_1 | grep IPAddress`
[sudo] password for ubuntu:
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "172.19.0.5",

2. Result of

ubuntu@ubuntu:~$  `sudo docker inspect exo_home_mysql_1 | grep IPAddress`
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "172.19.0.3",
ubuntu@ubuntu:~$

**Reply by rdenarie** [reply]
I quite puzzled about your problem.
The 2 container seems in the same network, but are unable to discuss.

So, I propose to change the docker-compose file to expose the port 3306 of the mysql server on your local machine, and configure exo to join directly your local machine instead of directly the mysql container.

For that, firsly, shutdown all the container : 
`docker-compose -f docker-compose.yml down`

Then, find your local ip (use command ifconfig). On my side, it is 192.168.0.2, so I'll take that as example. Change with your real IP
In docker-compose.yml, change line 
`      EXO_DB_HOST: mysql`
to 
`      EXO_DB_HOST: 192.168.0.2`
Then, in the mysql part, add these lines : 
>     ports:
>       - "3306:3306"

mysql part should be : 

> mysql:
>     image: mysql:8.0.28
>     environment:
>       MYSQL_ROOT_PASSWORD: my-super-secret-pw
>       MYSQL_DATABASE: exo
>       MYSQL_USER: exo
>       MYSQL_PASSWORD: my-secret-pw
>     volumes:
>       - mysql_data:/var/lib/mysql
>     ports:
>       - "3306:3306"

Save the docker file.

We will also drop all volumes already created. Find them with 
`docker volume ls |grep exo_home`
Then delete it : 
```
docker volume rm exo_home_exo_data
docker volume rm exo_home_exo_logs
docker volume rm exo_home_mongo_data
docker volume rm exo_home_mysql_data
docker volume rm exo_home_search_data
```

Finally, restart all containers : 
`docker-compose -f docker-compose.yml up`

After that, if it still not working, can you provide me the docker-compose.yml file, and all the starting log ?

Thanks

**Reply by BivanOdima** [reply]
Hello,

After exposing mysql port and running `docker-compose -f docker-compose.yml up`
 It brought a mongo db error. I exposed mongo db too and changed the exo_mongo_host to my IP. It was able to connect to both Mongo and MySql. 
However its now unable to connect to the Elastic Search Container Port.

Looks like we also have to expose that port.

I have attached a start up log
[startup log exo.txt](https://github.com/exoplatform/exo-community/files/9614359/startup.log.exo.txt)
[startup log exo.txt](https://github.com/exoplatform/exo-community/files/9614402/startup.log.exo.txt)

**Reply by rdenarie** [reply]
You're right, I think we should expose es port.
For that, modify the docker-compose.yml by changing 

>       EXO_ES_HOST: es
to

>       EXO_ES_HOST: 192.168.0.2

an in es part, add ports to expose : 

>     ports:
>       - "9200:9200"
>       - "9300:9300"

Then, restart your containers with 
```
docker-compose -f docker-compose.yml down
docker-compose -f docker-compose.yml up
```

**Comment by BivanOdima** [comment]
It looks like all containers have started as shown below.

ubuntu@ubuntu:~/EXO_HOME$ `sudo docker ps`
[sudo] password for ubuntu:

**CONTAINER ID   IMAGE                             COMMAND                  CREATED          STATUS                    PORTS                                                                                  NAMES
15246881b77a   exoplatform/elasticsearch:2.0.3   "/bin/tini -- /usr/l…"   42 minutes ago   Up 42 minutes             0.0.0.0:9200->9200/tcp, :::9200->9200/tcp, 0.0.0.0:9300->9300/tcp, :::9300->9300/tcp   exo_home_es_1
f4c5afcd513a   exoplatform/nginx:1.21.0          "/usr/bin/supervisord"   54 minutes ago   Up 42 minutes             81/tcp, 0.0.0.0:80->80/tcp, :::80->80/tcp, 443/tcp                                     exo_home_web_1
49a56f5c55f0   exoplatform/exo-community:6.3     "/usr/local/bin/tini…"   54 minutes ago   Up 42 minutes (healthy)   0.0.0.0:8080->8080/tcp, :::8080->8080/tcp                                              exo_home_exo_1
68711671d713   mongo:4.4                         "docker-entrypoint.s…"   54 minutes ago   Up 42 minutes             0.0.0.0:27017->27017/tcp, :::27017->27017/tcp                                          mongo
10bbd994b9a0   mysql:8.0.28                      "docker-entrypoint.s…"   54 minutes ago   Up 42 minutes             0.0.0.0:3306->3306/tcp, :::3306->3306/tcp, 33060/tcp                                   exo_home_mysql_1**

However, I am trying to access it using the server ip from the browser but I cant http://192.168.0.150. The Server ip is 192.168.0.150 for my Ubuntu server 22.04 LTS with no GUI.

Running the below ip of the exo_home_exo_1 and  exo_home_mysql_1


**ubuntu@ubuntu:~/EXO_HOME$` sudo docker inspect exo_home_exo_1 | grep IPAddress`
[sudo] password for ubuntu:
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "172.21.0.5",
ubuntu@ubuntu:~/EXO_HOME$` sudo docker inspect exo_home_mysql_1 | grep IPAddress`
            "SecondaryIPAddresses": null,
            "IPAddress": "",
                    "IPAddress": "172.21.0.2",**

**Reply by BivanOdima** [reply]
The following is the start up log.
[log.txt](https://github.com/exoplatform/exo-community/files/9615723/log.txt)

**Comment by rdenarie** [comment]
Can you try to access on http://192.168.0.150:8080 ?
It will bypass the frontend nginx.

There is a network issue with the docker, and I dont understand why for now. So I guess nginx can not communicate with eXo like eXo cannot communicate directly with mongo or es or mysql.

**Reply by BivanOdima** [reply]
Hello I managed to access exo succesfully. Port 8080 is working fine.

I have attached my docker-compose.yml.txt file
[docker-compose.yml.txt](https://github.com/exoplatform/exo-community/files/9624964/docker-compose.yml.txt)



Thank you very much.

**Reply by rdenarie** [reply]
Great news ! 

Thanks for your feedback.

**Reply by PIAODEV** [reply]
Hello, I tried my best to follow these responses, but my installation is still not working.

When accessing the localhost, I am only seeing the "Welcome to nginx!" screen.

I have uploaded my cmd lines.
[cmd.txt](https://github.com/exoplatform/exo-community/files/9660109/cmd.txt)

Also, I modified our docker-compose.yml file to mimic BivanOdima's. Do you have any suggestions?

**Reply by PIAODEV** [reply]
I think the issue somehow has to do with this: 
![image](https://user-images.githubusercontent.com/114536434/192657835-cec00320-2b99-4255-99aa-f6fcdef04fa2.png)

The web-1 seems to bypass the EXO Platform and direct straight to nginx:1.21.0 - could it be a folder issue by any chance?

**Reply by rdenarie** [reply]
Hello @PIAODEV 

Can you start a new discussion to describe your problem ? 
In it, can you provided the modified docker compose ? and the startup logs ?

Thanks a lot.

**Reply by PIAODEV** [reply]
Thank you @rdenarie - please see [this discussion](https://github.com/exoplatform/exo-community/discussions/6). I appreciate you help.

**Comment by BivanOdima** [comment]
I have

Bivan Odima (ICT)​​
Systems Administrator
Madison Group Limited
+254 020 2864000, +254 709 922 000
www.madison.co.ke<http://www.madison.co.ke/>
[Facebook]<https://www.facebook.com/MadisonGroupKenya/>
[LinkedIn]<https://www.linkedin.com/company/madison-group-ltd/>
[Twitter]<https://twitter.com/MadisonGroupKe>
[YouTube]<https://www.youtube.com/user/MadisonInsurance>
[Instagram]<https://www.instagram.com/madisongroup_ke/>
***@***.***
